### PR TITLE
Update query ability - can now query on Extra Data

### PR DESCRIPTION
**Pull request by blundell**
**breaking API change** need to remember this when we write the changelog & release

Fixes https://github.com/novoda/download-manager/pull/88#issuecomment-120435573

Now that we have an extra data column where clients can pass whatever they want through the download manager. It is right to also add the ability to query on this field.

Did some boyscouting and added 2 more tests and some clean up.

### CODE REVIEW

**Comment by hal9002** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.novoda.com/job/download-manager-prb/102/
Test PASSed.

**Comment by xrigau** [comment]
cc @Electryc

File: library/src/main/java/com/novoda/downloadmanager/lib/Query.java [hunk 1]
```diff
@@ -75,12 +77,22 @@ public Query setFilterByBatchId(long... batchIds) {
...
 
     /**
-     * Include only the downloads with the given extras.
+     * Include only the downloads with the given notification extras.
      *
      * @return this object
      */
-    public Query setFilterByExtras(String... extras) {
-        filterExtras = extras;
+    public Query setFilterByNotificationExtras(String... extras) {
```
Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/Query.java]: Worth mentioning this is a breaking API change :tongue:

File: library/src/test/java/com/novoda/downloadmanager/lib/QueryTest.java [hunk 2]
```diff
@@ -60,29 +64,76 @@ public void whenWeSetABatchStatusOrderByOnAQueryThenTheResolverIsQueriedWithTheC
...
+    }
+
+    @Test
+    public void givenNoExtraDataWhenTheQueryIsCreatedThenTheWhereStatementContainsNoBatchIdPredicate() {
+        query.runQuery(resolver, null, uri);
+
+        verify(resolver).query(any(Uri.class), any(String[].class), stringArgumentCaptor.capture(), any(String[].class), anyString());
+
+        assertSelectionDoesNotContain(COLUMN_EXTRA_DATA + " IN ");
+    }
```
Comment (Electryc) [inline library/src/test/java/com/novoda/downloadmanager/lib/QueryTest.java]: :100: the tests

**Comment by Electryc** [comment]
LGTM :+1:

File: library/src/main/java/com/novoda/downloadmanager/lib/Query.java (hunk 1 above)
Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/Query.java]: it is!

Comment (xrigau) [inline library/src/main/java/com/novoda/downloadmanager/lib/Query.java]: see updated readme

Comment (Electryc) [inline library/src/main/java/com/novoda/downloadmanager/lib/Query.java]: :+1:

**Comment by Electryc** [comment]
Let's :rocket: :ship: this

**Pull request merged**